src/dftbp/dftbplus/initprogram.F90

```diff
@@ -3880,6 +3891,54 @@ subroutine initProgramVariables(this, input, env)
       end do
     end if
 
+    if (allocated(this%apiCallBack)) then
+      if (this%apiCallBack%isAsiChangingTheModel()) then
+        if (allocated(this%scc)) then
+          ! as this needs assurances that the DM is being read by the external code, leading to
+          ! changes in the hamiltonian, otherwise SCC never converges.
+          call error("ASI callback with model modification enabled does not support self-consistent&
+              & calculations at present")
+        end if
+        if (this%tForces) then
+          ! Since if H and/or S is modified, the derivatives are not available via ASI at the
+          ! moment.
+          call error("ASI callback with model modification enabled does not support forces at&
+              & present")
+        end if
+        if (this%tMulliken) then
```

Comment:
This is only a problem if the overlap is modified, the hamiltonian can be changed safely for Mulliken.

Reply:
Testing the result of set_s_callback instead of just whether it is registered would be better.